File: Homework/Son/W1/922-sort-array-by-parity-II.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
 var sortArrayByParityII = function(nums) {
    let i = 0;
    let j = 1;
    while (i < nums.length -1 || j < nums.length -1 ) {
        if (nums[i] % 2 === 0) {i = i + 2;}
        if (nums[j] % 2 === 1) j+=2;
        if (nums[i] % 2 === 1 && nums[j] % 2 === 0) {
            let temp = nums[i];
            nums[i] = nums[j];
            nums[j] = temp;
            i+=2;
            j+=2;
        }
    }
    return nums;
};

// https://leetcode.com/problems/sort-array-by-parity-ii/

File: Homework/Ngan_Luu/922-sort-array-by-parity-ii.py
#https://leetcode.com/problems/sort-array-by-parity-ii/ (O(1) extra space)
class Solution:
    def sortArrayByParityII(self, nums: List[int]) -> List[int]:
        p=0
        i=1
        while p<len(nums) and i<len(nums):
            if nums[p]%2==0:
                p+=2
            elif nums[i]%2==0:
                nums[p],nums[i]=nums[i],nums[p]
                p+=2
                i+=2
            else:
                i+=2
        return nums